inc/container.class.php

```diff
@@ -1962,7 +1962,9 @@ public static function getAddSearchOptions($itemtype, $containers_id = false)
                     $opt[$i]['datatype'] = 'text';
                     break;
                 case 'number':
-                    $opt[$i]['datatype'] = 'decimal';
+                    // change datatype to string to get `is` / `is not` operator
+                    $opt[$i]['datatype'] = 'string';
+                    $opt[$i]['searchtype'] = ['contains', 'notcontains', 'equals', 'notequals'];
```

Comment:
Just adding `searchtype` wouldn't be enough?
```suggestion
                    $opt[$i]['datatype'] = 'decimal';
                    $opt[$i]['searchtype'] = ['contains', 'notcontains', 'equals', 'notequals'];
```

Reply:
decimal type have a different behavior from Search.class to handle missing `.`

                if (isset($searchopt[$ID]["datatype"]) && ($searchopt[$ID]["datatype"] === 'decimal')) {
                    $matches = [];
                    if (preg_match('/^(\d+.?\d?)/', $val, $matches)) {
                        $val = $matches[1];
                        if (!str_contains($val, '.')) {
                            $val .= '.';
                        }
                    }
                }

Sa we cannot handle all type of search; knowing that this field is in text format, and can be used to save numeric or decimal fields (a nightmare)

Reply:
in my opinion, with your change, not all of the following conditions will be functional

is 15 (match only 15 not 15.50 or other)
is not 15 (match other expect 15)
contain 15 (match 15.10, 10.15, 15)
not contain 15 (match other than 15.10, 10.15, 15)
>= 15 (match 15.05 , 15, 16)
<= 15 (match 10, ,10.50, 15)